<launch>
  <rosparam file="$(find joy_control_sample_ros)/config/joy_control.yaml"/>
  <rosparam file="$(find joy_control_sample_ros)/config/cbf.yaml"/>

  <node name="joy" pkg="joy" type="joy_node"/>
  <node name="cbf_optimizer" pkg="joy_control_sample_ros" type="cbf_optimizer.py" output="screen"/>

  <node name="controller" pkg="joy_control_sample_ros" type="controller.py">
    <remap from="cmd_vel" to="cmd_vel_nom"/>
  </node>
  <node name="agent_body" pkg="joy_control_sample_ros" type="agent_body.py">
    <remap from="cmd_vel" to="cmd_vel_opt"/>
  </node>
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find joy_control_sample_ros)/rviz/joy_control.rviz"/>
</launch>